app/models/job.ts

```diff
@@ -526,6 +539,30 @@ export class Job extends Record implements JobRecord {
     return data.length !== 0;
   }
 
+  /**
+   * Determine if the current job is the same as an already running job. If so, replace the
+   * current job's requestId with that of the running job's. Add an attachedStatus property for
+   * later referencing.
+   *
+   * @param transaction - The transaction to use for saving the job
+   * @throws {@link Error} if the job is invalid
+   */
+  async maybeAttach(transaction: Transaction): Promise<void> {
+    const rows = await transaction('jobs')
+      .select('requestId', 'shapeFileHash')
+      .where({ username: this.username, request: this.request, shapefileHash: this.shapeFileHash })
```

Comment:
Checking for `this.request` should work for GET requests, but I'm worried that for POST requests we're going to consider different requests as the same request because we're not tracking the form fields.